paderbox/array/sparse.py

```diff
@@ -0,0 +1,701 @@
+import bisect
+import copy
+from dataclasses import dataclass, field
+
+from typing import Tuple, Any
+
+from cached_property import cached_property
+
+from paderbox.array.interval.core import ArrayInterval
+import numpy as np
+import paderbox as pb
+from paderbox.array.interval.util import cy_parse_item, cy_invert_intervals
+import torch
+
+
+def _normalize_shape(shape):
+    if shape is None:
+        # As of now, we only support fixed shapes. This can be changed
+        raise TypeError(f'None shape is not supported')
+
+    if isinstance(shape, int):
+        return shape,
+
+    if not isinstance(shape, (tuple, list)):
+        raise TypeError(f'Invalid shape {shape} of type {type(shape)}')
+
+    if not all(isinstance(s, int) for s in shape):
+        raise TypeError(
+            f'Invalid shape {shape} with elements of type {type(shape[0])}'
+        )
+
+    return tuple(shape)
+
+
+def zeros(shape):
+    return SparseArray(shape=shape)
+
+
+def _check_shape(shape1, shape2):
+    if shape1 != shape2:
+        raise ValueError(f'Shape mismatch: {shape1} {shape2}')
+
+
+def _clip(target, array, onset):
```

Comment:
I found `_clip` confusing (`np.clip`). Is cat a better name?

Reply:
I think you mean `_cut`, but I agree it would be a better fit than `_clip`.